SU2_CFD/include/numerics/turbulent/turb_sources_new.hpp

```diff
@@ -143,520 +143,373 @@ class CSourceBase_TurbSA : public CNumerics {
 
-      //    Original SA model
-      //    Destruction = (cw1*fw-cb1*ft2/k2)*ScalarVar_i[0]*ScalarVar_i[0]/dist_i_2*Volume;
+      /*--- Residual ---*/
+      Residual = Production - Destruction + CrossProduction + AddSourceTerm;
+      Residual *= Volume;
 
-      Destruction = cw1 * fw * ScalarVar_i[0] * ScalarVar_i[0] / dist_i_2 * Volume;
+      /*--- Jacobian ---*/
+      Jacobian_i[0] *= Volume;
+    }
+  }
+};
 
-      /*--- Diffusion term ---*/
+/*------------------------------------------------------------------------------
+| Structure with SA common auxiliary functions and constants
+------------------------------------------------------------------------------*/
+struct CommonVariables {
+   /*--- List of constants ---*/
+   su2double cv1_3, k2, cb1, cw2, ct3, ct4, cw3_6, cb2_sigma, sigma, cb2, cw1, cr1;
 
-      norm2_Grad = 0.0;
-      for (iDim = 0; iDim < nDim; iDim++) norm2_Grad += ScalarVar_Grad_i[0][iDim] * ScalarVar_Grad_i[0][iDim];
+   /*--- List of auxiliary functions ---*/
+   su2double ft2, d_ft2, r, d_r, g, d_g, glim, fw, d_fw, Ji, d_Ji, S, Shat, d_Shat, fv1, d_fv1, fv2, d_fv2;
 
-      CrossProduction = cb2_sigma * norm2_Grad * Volume;
+   /*--- List of helpers ---*/
+   su2double Omega, dist_i_2, inv_k2_d2, inv_Shat, g_6, norm2_Grad;
+};
 
-      Residual = Production - Destruction + CrossProduction;
+/*------------------------------------------------------------------------------
+| Strain rate
+------------------------------------------------------------------------------*/
 
-      /*--- Implicit part, production term ---*/
+/*!
+ * \brief Baseline
+ */
+struct Omega_Bsl {
+   template <class Base>
+   static void get(CommonVariables &model_var) {
 
-      dfv1 = 3.0 * Ji_2 * cv1_3 / (nu * pow(Ji_3 + cv1_3, 2.));
-      dfv2 = -(1 / nu - Ji_2 * dfv1) / pow(1. + Ji * fv1, 2.);
-      if (Shat <= 1.0e-10)
-        dShat = 0.0;
-      else
-        dShat = (fv2 + ScalarVar_i[0] * dfv2) * inv_k2_d2;
+      using Base::Vorticity_i;
 
-      if (transition) {
-        Jacobian_i[0] += Gamma_BC * cb1 * (ScalarVar_i[0] * dShat + Shat) * Volume;
-      } else {
-        Jacobian_i[0] += cb1 * (ScalarVar_i[0] * dShat + Shat) * Volume;
-      }
+      model_var.Omega = sqrt(Vorticity_i[0] * Vorticity_i[0] + Vorticity_i[1] * Vorticity_i[1] + Vorticity_i[2] * Vorticity_i[2]);
+   }
+};
 
-      /*--- Implicit part, destruction term ---*/
+/*!
+ * \brief Edward
+ *
+ * Here Omega is the Strain Rate
+ */
+struct Omega_Edw {
+   template <class Base>
+   static void get(CommonVariables &model_var) {
 
-      dr = (Shat - ScalarVar_i[0] * dShat) * inv_Shat * inv_Shat * inv_k2_d2;
-      if (r == 10.0) dr = 0.0;
-      dg = dr * (1. + cw2 * (6.0 * pow(r, 5.0) - 1.0));
-      dfw = dg * glim * (1. - g_6 / (g_6 + cw3_6));
-      Jacobian_i[0] -= cw1 * (dfw * ScalarVar_i[0] + 2.0 * fw) * ScalarVar_i[0] / dist_i_2 * Volume;
+      using Base::PrimVar_Grad_i;
+      using Base::idx;
```

Comment:
Looking good @suargi :+1:, thanks for defining the common vars struct already.

If you template `get` on `Base` you need to pass something to `get` from which `Base` can be deduced, and the `using` declarations here will not give access to those variables.
Best solution IMO, do not template on base (or anything else to keep it simple) and pass what you need to `get`.

Reply:
Thanks. In 123302d8acb96c34e672afb428129d1da3a8b4ca I have restructured the `get` methods to only pass those necessary variables instead of the class.